app/core/auth_jwt_bearer.py

```diff
@@ -21,13 +20,18 @@
 
 def decode_jwt(token: str) -> Optional[dict]:  # noqa: UP007
     """
-    this function decodes a JWT token
-    and validates time and issuer/audience claims.
-
-    :param token:
-    :type token:
-    :return:
-    :rtype:
+    Decodes a JWT token and validates its critical claims,
+    including time-based and issuer/audience claims.
+
+
+    :param token: The JWT token to decode.
+    :return: The decoded token as a dictionary if valid, or `None`
+    if the token is invalid or expired.
+    :raises:
+        ExpiredSignatureError: If the token's signature has expired.
+        DecodeError: If the token cannot be decoded due to formatting or cryptographic issues.
+        InvalidTokenError: If the token is otherwise invalid.
+        Exception: For any general decoding errors.
```

Comment:
> Well, I like writing docs and examples

Oh fantastic then! Documentation is always good to have, hard to find people who enjoy writing it

I suppose then my only worry is that the code style may get too mixed as we start collaborating on the same projects. This means that for a new joiner it would be harder to navigate and understand, however there are solutions to this  and we can get to it if it actually becomes a problem